opal/mca/common/ofi/common_ofi.c

```diff
@@ -445,63 +445,214 @@ static int check_provider_attr(struct fi_info *provider_info, struct fi_info *pr
     }
 }
 
-#if OPAL_OFI_PCI_DATA_AVAILABLE
-/* Check if a process and a pci device share the same cpuset
- *     @param (IN) pci              struct fi_pci_attr pci device attributes,
- *                                  used to find hwloc object for device.
+/**
+ * Calculate device distances
+ *
+ * Calculate the distances between the current thread and all devices of
+ * type OPENFABRICS or NETWORK.
+ *
+ * The shortest distances are the nearest and therefore most efficient
+ * devices to use.
  *
- *     @param (IN) topology         hwloc_topology_t topology to get the cpusets
- *                                  from
+ * Return an array of all the distances. Each entry is of type
+ * pmix_device_distance_t
  *
- *     @param (OUT)                 returns true if cpusets match and false if
- *                                  cpusets do not match or an error prevents comparison
+ * This function is used if there is no PMIx server running.
+ *
+ * @param distances (OUT)     distances array
+ * @param ndist (OUT)    number of entries in the distances array
+ *
+ * @return   0 on success. Error otherwise.
  *
- *     Uses a pci device to find an ancestor that contains a cpuset, and
- *     determines if it intersects with the cpuset that the process is bound to.
- *     if the process is not bound, or if a cpuset is unavailable for whatever
- *     reason, returns false. Otherwise, returns the result of
- *     hwloc_cpuset_intersects()
  */
-static bool compare_cpusets(hwloc_topology_t topology, struct fi_pci_attr pci)
+static int calculate_distances(pmix_device_distance_t **distances,
```

Comment:
Maybe more appropriately named to discover_devices_and_calc_distances or something

Reply:
renamed to compute_dev_distances() as we're not really discovering devices. HWLOC has already discovered the topology. We're just using it to calculate the distance to the devices.